File: dotnet/bindings/CLRInterface/ClassField.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DNP3.Interface
{
    /// <summary>
    /// Structure that records which events are scanned / evented
    /// </summary>
    public struct ClassField
    {      
        public ClassField(byte mask)
        {
            this.bitfield =  mask;
        }

        public static ClassField From(params PointClass[] classes)
        {
            byte mask = 0;
            foreach(var pc in classes)
            {
                mask |= (byte)pc;
            }
            return new ClassField(mask);
        }

        public static ClassField AllClasses
        {
            get
            {
                return ClassField.From(PointClass.Class0, PointClass.Class1, PointClass.Class2, PointClass.Class3);
            }
        }

        public static ClassField AllEventClasses
        {
            get
            {
                return ClassField.From(PointClass.Class1, PointClass.Class2, PointClass.Class3);
            }
        }

        public byte ClassMask
        {
            get
            {
                return bitfield;
            }
        }

        public  bool HasClass0
        {
            get
            {
                return (((byte)PointClass.Class0) & bitfield) != 0;
            }
        }

        public bool HasClass1
        {
            get
            {
                return (((byte)PointClass.Class1) & bitfield) != 0;
            }
        }

        public bool HasClass2
        {
            get
            {
                return (((byte)PointClass.Class2) & bitfield) != 0;
            }
        }

        public bool HasClass3
        {
            get
            {
                return (((byte)PointClass.Class3) & bitfield) != 0;
            }
        }

        private byte bitfield;
    }   
}

File: java/api/src/main/java/com/automatak/dnp3/ClassField.java
package com.automatak.dnp3;


import com.automatak.dnp3.enums.PointClass;

/**
 * Structure that records which events are scanned / evented
 */
public class ClassField {

    private int bitfield;

    private ClassField(int mask) {
        this.bitfield = mask;
    }

    public static ClassField from(PointClass... classes) {
        byte mask = 0;
        for (PointClass pc : classes) {
            mask |= pc.toType();
        }
        return new ClassField(mask);
    }

    public boolean isSet(PointClass pc)
    {
        return (pc.toType() & bitfield) != 0;
    }


    public static ClassField none() {
        return new ClassField(0);
    }

    public static ClassField allClasses() {
        return new ClassField(PointClassMasks.ALL_CLASSES);
    }

    public static ClassField allEventClasses() {
        return new ClassField(PointClassMasks.ALL_EVENTS);
    }

}
